iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergOutputCommitter.java

```diff
@@ -481,7 +484,23 @@ private void commitTable(FileIO io, ExecutorService executor, OutputTable output
           .map(x -> x.getJobConf().get(ConfVars.REWRITE_POLICY.varname))
           .orElse(RewritePolicy.DEFAULT.name()));
 
-      commitOverwrite(table, branchName, startTime, filesForCommit, rewritePolicy);
+      Integer compactionPartSpecId = outputTable.jobContexts.stream()
+          .findAny()
+          .map(x -> x.getJobConf().get(IcebergCompactionContext.COMPACTION_PART_SPEC_ID))
+          .map(Integer::valueOf)
+          .orElse(null);
+
+      String compactionPartitionPath = outputTable.jobContexts.stream()
```

Comment:
could we please use consistent naming, not
````
compactionPartSpecId/COMPACTION_PART_SPEC_ID
compactionPartitionPath/COMPACTION_PARTITION_PATH
````
why do we need compaction in name? how about:
````
partitionSpecId/PARTITION_SPEC_ID
partitionPath/PARTITION_PATH
````

Reply:
also why don't we compute path only inside compaction block?

Reply:
Done